src/ir/IRInlineHandler.ts

```diff
@@ -1,82 +1,102 @@
 import { DisallowedArgError, ErrorLogger } from "..";
 import { InlineHandler, Sugar, SugarsByStart } from "../parser/InlineHandler";
 import { InputPosition } from "../parser/InputPosition";
-import { HMError } from "../schema/errors";
+import {
+	DisallowedInArgError,
+	DisallowedInHeadError,
+	UnknownInlineTagError
+} from "../schema/errors";
 import { InlineSchema, Schema } from "../schema/Schema";
+import { InlinePropDefinition } from "../schema/SchemaDefinition";
 import { last } from "../utils";
 import { emptyBlockProps, IRNode, IRNodeList } from "./IRNode";
 
 export class IRInlineHandler implements InlineHandler {
-	protected readonly inlineGroupStack: IRNodeList[] = [];
-	protected readonly inlineElementStack: Array<{ node: IRNode; schema: InlineSchema }> = [];
-	protected ignoreFlag: boolean = false;
+	protected readonly inlineGroupStack: Array<{
+		nodeList: IRNodeList;
+		schema: InlinePropDefinition;
+	} | null> = [];
+	protected readonly inlineElementStack: Array<{
+		node: IRNode;
+		schema: InlineSchema;
+	} | null> = [];
 
-	constructor(private schema: Schema, private readonly logger: ErrorLogger) {}
+	constructor(private schema: Schema, private readonly log: ErrorLogger) {}
 
 	getResult(): IRNodeList {
-		return this.inlineGroupStack[0];
+		return this.inlineGroupStack[0]!.nodeList;
 	}
 
-	reset() {
+	reset(rootSchema: InlinePropDefinition) {
 		this.inlineGroupStack.length = 0;
 		this.inlineElementStack.length = 0;
-		this.inlineGroupStack.push([]);
-		this.ignoreFlag = false;
+		this.inlineGroupStack.push({ nodeList: [], schema: rootSchema });
 	}
 
 	openInlineTag(tag: string, pos: InputPosition): void {
-		if (this.ignoreFlag) return;
 		const parent = last(this.inlineGroupStack);
+		if (!parent) {
+			this.inlineElementStack.push(null);
+			return;
+		}
 		const schema = this.schema.getInlineSchema(tag);
-		if (!schema) throw new Error("blop");
-		// TODO: check that tag is valid child of parent tag
-		// TODO: check that tag is known inline
+		if (!schema) {
+			this.log(new UnknownInlineTagError(tag, pos));
+			return;
+		}
+
+		if (parent.schema.raw) throw new Error("");
+
+		if (!parent.schema.content.has(tag)) {
+			if (this.inlineElementStack.length > 0) {
+				this.log(new DisallowedInArgError(tag, parent.schema.name, pos));
+			} else {
+				this.log(new DisallowedInHeadError(tag, "[parent]", pos));
```

Comment:
`"[parent]"` should be `parent.schema.tag`, but that's not possible currently. See my comment above about `ArgSchema` for a possible solution